xarray/backends/common.py

```diff
@@ -264,19 +264,22 @@ def add(self, source, target):
         else:
             target[...] = source
 
-    def sync(self):
+    def sync(self, compute=True):
         if self.sources:
             import dask.array as da
-            da.store(self.sources, self.targets, lock=self.lock)
+            delayed = da.store(self.sources, self.targets, lock=self.lock,
+                               compute=compute, flush=True)
             self.sources = []
             self.targets = []
+            return delayed
```

Comment:
It may be wise to use another term here to avoid confusion with the `dask.delayed` function itself.